src/pybind/mgr/rgw/module.py

```diff
@@ -349,3 +349,54 @@ def shutdown(self) -> None:
         self.log.info('Stopping')
         self.run = False
         self.event.set()
+
+    def list_all_realm_tokens(self) -> List[Dict]:
+        realms_info = []
+        for realm_info in RGWAM(self.env).get_realms_info():
+            if not realm_info['master_zone_id']:
+                realms_info.append({'realm': realm_info['realm_name'], 'token': 'realm has no master zone'})
+            elif not realm_info['endpoint']:
+                realms_info.append({'realm': realm_info['realm_name'], 'token': 'master zone has no endpoint'})
+            elif not (realm_info['access_key'] and realm_info['secret']):
+                realms_info.append({'realm': realm_info['realm_name'], 'token': 'master zone has no access/secret keys'})
+            else:
+                keys = ['realm_name', 'realm_id', 'endpoint', 'access_key', 'secret']
+                realm_token = RealmToken(**{k: realm_info[k] for k in keys})
+                realm_token_b = realm_token.to_json().encode('utf-8')
+                realm_token_s = base64.b64encode(realm_token_b).decode('utf-8')
+                realms_info.append({'realm': realm_info['realm_name'], 'token': realm_token_s})
+        return realms_info
+
+    def import_realm_token(self,
+                           zone_name: Optional[str] = None,
+                           realm_token: Optional[str] = None,
+                           port: Optional[int] = None,
+                           placement: Optional[str] = None,
+                           start_radosgw: Optional[bool] = True,
+                           zone_endpoints: Optional[str] = None) -> str:
+        if (zone_name and realm_token):
+            token = RealmToken.from_base64_str(realm_token)
+            placement_spec = PlacementSpec.from_string(placement) if placement else None
+            rgw_specs = [RGWSpec(rgw_realm=token.realm_name,
+                                 rgw_zone=zone_name,
+                                 rgw_realm_token=realm_token,
+                                 rgw_frontend_port=port,
+                                 placement=placement_spec,
+                                 zone_endpoints=zone_endpoints)]
+        else:
+            err_msg = 'Please provide the zone_name and realm_token.'
+            return err_msg
+            # return HandleCommandResult(retval=-errno.EINVAL, stdout='', stderr=err_msg)
+
+        try:
+            created_zones = []
+            for rgw_spec in rgw_specs:
+                RGWAM(self.env).zone_create(rgw_spec, start_radosgw)
+                if rgw_spec.rgw_zone is not None:
+                    created_zones.append(rgw_spec.rgw_zone)
+        except RGWAMException as e:
+            return e.message
+            # self.log.error('cmd run exception: (%d) %s' % (e.retcode, e.message))
+            # return HandleCommandResult(retval=e.retcode, stdout=e.stdout, stderr=e.stderr)
+
+        return f"Zones {', '.join(created_zones)} created successfully"
```

Comment:
This code duplicates the same functionality (literally) as the method (below). Why can't you use the below method? is it because of the decorators? 

```
    def _cmd_rgw_zone_create(self,
                             zone_name: Optional[str] = None,
                             realm_token: Optional[str] = None,
                             port: Optional[int] = None,
                             placement: Optional[str] = None,
                             start_radosgw: Optional[bool] = True,
                             zone_endpoints: Optional[str] = None,
                             inbuf: Optional[str] = None) -> HandleCommandResult:

```

Reply:
Just forgot to refactor this..Did it now